### PR TITLE
Extendedrequests

### PR DESCRIPTION
**Pull request by levilime**
We can now accept or deny requests to buy our land and requests are cleaned.

### CODE REVIEW

File: Municipality/tygronEvents.mod2g [hunk 1]
```diff
@@ -84,6 +85,13 @@ module tygronEvents {
 		member(request("INFORMATION", RequestType, _, ContentLinkID, [SenderID|_], _, _, _,  _, _), List), (AreaSize == "NO_SIZE"; Price == "NO_PRICE"))
 		do insert(open_request(RequestType, ID, ContentLinkID, SenderID, ActionlogIDs, Price, Multipolygon, AreaSize, AnswerList)).
 		
+	%Only put those requests in the belief base that we need to answer.
+	forall percept(requests(List)), bel(member(request("INTERACTION_WITH_DATE", "PERMIT", ID, ContentLinkID, _, ActionlogIDs, Price, Multipolygon, AreaSize, AnswerList),List),
+		member(request("INFORMATION", "PERMIT", _, ContentLinkID, [SenderID|_], _, _, _,  _, _), List))
+		do insert(open_request("PERMIT", ID, ContentLinkID, SenderID, ActionlogIDs, Price, Multipolygon, AreaSize, AnswerList)).
+		
+	if percept(requests(List)) then print(requests(List)).
```
Comment (NickTUD) [inline Municipality/tygronEvents.mod2g]: Print line

File: Municipality/tygronEvents.mod2g [hunk 2]
```diff
@@ -94,6 +102,6 @@ module tygronEvents {
 	if true then negotiation.
 	
 	%Account for the indicator concerning green.
-	if true then green.
-	if true then azc.
+	%if true then green.
+	%if true then azc.
```
Comment (NickTUD) [inline Municipality/tygronEvents.mod2g]: Why is this commented?

File: Municipality/tygron.mod2g [hunk 3]
```diff
@@ -8,23 +8,21 @@ exit=never.
...
-		then building_plan_construction(FunctionID, 1, multipolygon('MULTIPOLYGON (((10 20, 10 30, 210 30, 210 20, 10 20)))')).
-	
-	% If there is a function to build a road
-	% Then use building_plan_construction to build it (always 1 floor)
-	if bel(functions(FunctionList), member([_,FunctionID,List],FunctionList), member('ROAD',List)) 
-		then building_plan_construction(FunctionID, 1, multipolygon('MULTIPOLYGON (((10 30, 10 40, 210 40, 210 30, 10 30)))')).
-
-
+	%if goal(buildPark(ZoneID,MultiPolygon)), bel(functions(FunctionList), member(['stadspark',FunctionID,List],FunctionList)) 
+	%	then building_plan_construction(FunctionID, 1, MultiPolygon).
```
Comment (NickTUD) [inline Municipality/tygron.mod2g]: Shouldn't this be uncommented as well?

File: Municipality/tygronEvents.mod2g [hunk 4]
```diff
@@ -2,6 +2,7 @@ use dummy as knowledge.
 use green as module.
 use azc as module.
 use negotiation as module.
+use tygron as actionspec.
```
Comment (NickTUD) [inline Municipality/tygronEvents.mod2g]: For what purpose is the actionspec module added?

**Comment by NickTUD** [comment]
Okay, this looks fine by me :+1:

File: Municipality/tygron.mod2g [hunk 5]
```diff
@@ -8,23 +8,21 @@ exit=never.
...
 	if goal(buildPark(ZoneID,MultiPolygon)), bel(functions(FunctionList), member(['stadspark',FunctionID,List],FunctionList)) 
 		then building_plan_construction(FunctionID, 1, MultiPolygon).
-	
-	% If there is a goal to have an AZC, build an AZC.
-	if goal(buildAZC), bel(functions(FunctionList), member(['azc',FunctionID,List],FunctionList)) 
-		then building_plan_construction(FunctionID, 1, multipolygon('MULTIPOLYGON (((10 20, 10 30, 210 30, 210 20, 10 20)))')).
-	
-	% If there is a function to build a road
-	% Then use building_plan_construction to build it (always 1 floor)
-	if bel(functions(FunctionList), member([_,FunctionID,List],FunctionList), member('ROAD',List))
```
Comment (dionytadema) [inline Municipality/tygron.mod2g:26]: Please don't remove the rule for Building an azc

Comment (levilime) [inline Municipality/tygron.mod2g:26]: Building an AZC at a fixed polygon is not functional to me I don't see a problem in leaving it out.

**Comment by levilime** [comment]
Let's merge this ticket please

**Comment by StefanV7** [comment]
Looks cool :+1:

**Pull request merged**